### PR TITLE
feat - Compact sidebar

### LINKED ISSUE #7143: Compact sidebar
# Current behavior

Today, you can hide the sidebar. When the sidebar is hidden, all commands are also hidden. To navigate, you need to either open ⌘K or reopen the navbar.

CleanShot 2024-09-19 at 11 31 44

# Desired Behavior

Instead of hiding the full sidebar, we want to transform it into a "Compact sidebar" that displays only the navigation item icons. Please pay attention to UX details while implementing this 🙏

image

# Figma

https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=5910-77814&node-type=frame&t=GPMRKGqFNWATUOMD-11

### PR DESCRIPTION
**Pull request by ehconitin**
> fixes #7143 
> @Bonapara 

> https://github.com/user-attachments/assets/974702b5-acff-4d96-9742-02f2adbed620

Recent updates 

https://github.com/user-attachments/assets/7ff3dbc9-ff1b-447f-a7f0-ac0e920a3554

## To Do
- fix breadcrumb 
image

### CODE REVIEW

**Comment by ehconitin** [comment]
@Bonapara 
need clarifications on greyboxes border color and background color!
for now I have kept it according to figma  theme.background.transparent.lighter for both, but that could be improved

**Comment by FelixMalfait** [comment]
Nice 😍

**Comment by Bonapara** [comment]
@ehconitin what do you mean about the clarification on the colors? We can schedule a quick call to talk about it!

A few fixes:

CleanShot 2024-10-09 at 11 04 29

- Make sure the border-radius are the same
- The colors of the "views" icons should be tertiary to distinguish them from objects better. (including the selected view)

https://github.com/user-attachments/assets/dca0443f-34ad-434a-809b-da253c0e69d0

- The animation should be smooth like this. We shouldn't see that it's two different elements.
- The gap between sections in compact mode should be smaller.

**Comment by martmull** [comment]
@ehconitin think we can do something like on figma with animate. We need to animate all the disapearing components in the navBar, not the navBar itself. Here is a quick proof of concept: https://github.com/twentyhq/twenty/pull/7576/commits/de774637e9020ca78ddef123381343c64fb3eec6

https://github.com/user-attachments/assets/352e3cde-3bec-4196-9bec-7367141b160c

**Comment by martmull** [comment]
last version
https://github.com/user-attachments/assets/058a69d1-f5ad-401a-b964-0178bb0dc7ff

**Review by thomtrp — APPROVED** [review]

File: packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx [hunk 1]
```diff
@@ -78,7 +80,12 @@ const StyledItem = styled('div', {
     indentationLevel === 2 ? '2px' : '0'};
 
   pointer-events: ${(props) => (props.soon ? 'none' : 'auto')};
-  width: 100%;
+
+  width: ${(props) =>
+    !props.isNavigationDrawerExpanded
+      ? `${NAV_DRAWER_WIDTHS.menu.desktop.collapsed - 24}px`
```
[the remark this answers is not in the document]
Comment (martmull) [reply packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:86]: will try to

[the remark this answers is not in the document]
Comment (martmull) [reply packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx:86]: difficult tbh

**Pull request merged**